cryptofeed/pairs.py

```diff
@@ -324,59 +431,117 @@ def bitmax_pairs() -> Dict[str, str]:
+    r = None
+    try:
+        currencies = ['BTC', 'ETH']
+        kind = ['future', 'option']
+        data = []
+        for c in currencies:
+            for k in kind:
+                r = requests.get(f"https://www.deribit.com/api/v2/public/get_instruments?currency={c}&expired=false&kind={k}")
+                data.extend(r.json()['result'])
+        return {d['instrument_name']: d['instrument_name'] for d in data}
+    except Exception as why:
+        LOG.critical('DERIBIT: encountered %r while processing response from exchange API', why)
+        LOG.critical('DERIBIT: unexpected response: %r', r.text if r else r)
+        raise ValueError('Cryptofeed stopped because of an unexpected response from DERIBIT') from why
 
 
 def kraken_future_pairs() -> Dict[str, str]:
-    data = requests.get("https://futures.kraken.com/derivatives/api/v3/instruments").json()['instruments']
-    return {d['symbol']: d['symbol'] for d in data if d['tradeable'] is True}
+    r = None
+    try:
+        r = requests.get('https://futures.kraken.com/derivatives/api/v3/instruments')
+        data = r.json()['instruments']
+        return {d['symbol']: d['symbol'] for d in data if d['tradeable'] is True}
+    except Exception as why:
+        LOG.critical('KRAKEN_FUTURES: encountered %r while processing response from exchange API', why)
+        LOG.critical('KRAKEN_FUTURES: unexpected response: %r', r.text if r else r)
+        raise ValueError('Cryptofeed stopped because of an unexpected response from KRAKEN_FUTURES') from why
 
 
 def probit_pairs() -> Dict[str, str]:
-    r = requests.get("https://api.probit.com/api/exchange/v1/market").json()
-    return {entry['id']: entry['id'] for entry in r['data']}
+    # doc: https://docs-en.probit.com/reference-link/market
+    r = None
+    try:
+        r = requests.get('https://api.probit.com/api/exchange/v1/market')
+        return {entry['id']: entry['id'] for entry in r.json()['data']}
+    except Exception as why:
+        LOG.critical('PROBIT: encountered %r while processing response from exchange API', why)
+        LOG.critical('PROBIT: unexpected response: %r', r.text if r else r)
+        raise ValueError('Cryptofeed stopped because of an unexpected response from PROBIT') from why
 
 
 def coingecko_pairs() -> Dict[str, str]:
-    data = requests.get('https://api.coingecko.com/api/v3/coins/list').json()
-    return {entry['symbol'].upper(): entry['id'] for entry in data}
+    r = None
+    try:
+        r = requests.get('https://api.coingecko.com/api/v3/coins/list')
+        return {entry['symbol'].upper(): entry['id'] for entry in r.json()}
+    except Exception as why:
+        LOG.critical('COINGECKO: encountered %r while processing response from exchange API', why)
+        LOG.critical('COINGECKO: unexpected response: %r', r.text if r else r)
+        raise ValueError('Cryptofeed stopped because of an unexpected response from COINGECKO') from why
```

Comment:
since nearly all the error messages are the same format - exchange something something, could we just store the strings as global/consts and just format them in place in all the log lines? Then if ever the format needs to be slightly tweaked or adjusted, there is only one place to do it?

Reply:
Good idea :+1:
As you prefer, I can pool the strings myself or let you do it...
Please let me know :wink:

Reply:
Hey @bmoscon I am working on this requested improvement... :rocket: